### PR TITLE
auth/kubernetes: add changelog for issuer deprecation

### PR DESCRIPTION
**Pull request by imthaghost**
This adds a missing changelog in 1.9 for Kubernetes auth issuer deprecation: hashicorp/vault-plugin-auth-kubernetes#127.

The plugin was updated in the PR to Vault: #12975.

### CODE REVIEW

**Review by mladlow — CHANGES_REQUESTED** [review]
Thanks for adding this! If this PR merges to main this note will appear in the 1.10 CL, but not the 1.9 CL. If we'd like this to appear in the 1.9 CL, this PR needs to merge to the 1.9 release branch _only_, and not the main branch.

File: changelog/13126.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:deprecation
+auth/kubernetes: `disable_iss_validation` and `issuer` configuration fields deprecation  [#127](https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/127)
```
Comment (tvoran) [inline changelog/13126.txt:2]: Just a thought on wording:

```suggestion
auth/kubernetes: deprecate `disable_iss_validation` and `issuer` configuration fields [#127](https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/127)
```

File: changelog/13126.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+```
```
Comment (tvoran) [inline changelog/13126.txt:3]: I'm thinking this should have a `change` entry too, in the same file, something like:

````
```release-note:change
auth/kubernetes: default `disable_iss_validation` to true [[GH-127](https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/127)]
``` 
````

**Comment by mladlow** [comment]
Since this hasn't merged, and we've staged the release, we should plan to make changelog updates directly on the changelog itself. I'm happy to chat through the whys of that in more depth.

**Pull request closed without merging**